microscope/monitor/runner.py

```diff
@@ -240,7 +250,19 @@ def labels_match(data, selectors: List[str],
                 continue
             break
 
-        ids.update(namesMatch, labelsMatch)
+        ipsMatch = {
+            endpoint['id'] for endpoint in data
+            if any(
+                ip in
+                (item for sublist in
+                 [x.values() for x in
+                  endpoint['status']['networking']['addressing']
+                  ] for item in sublist)
+                for ip in ips
+            )
+        }
```

Comment:
This is finding all the IPs associated with an endpoint, right? I don't think I understand the comprehension(s) but in any case, this information is in the resolver object as an ip->endpoint map. That's probably more clear in intent?